pkg/kubelet/status/status_manager.go

```diff
@@ -566,43 +585,44 @@ func (m *manager) syncBatch() {
 			}
 		}
 
+		// Calculate all possible status updates
 		for uid, status := range m.podStatuses {
-			syncedUID := kubetypes.MirrorPodUID(uid)
-			if mirrorUID, ok := podToMirror[kubetypes.ResolvedPodUID(uid)]; ok {
-				if mirrorUID == "" {
-					klog.V(5).InfoS("Static pod does not have a corresponding mirror pod; skipping",
-						"podUID", uid,
-						"pod", klog.KRef(status.podNamespace, status.podName))
-					continue
-				}
-				syncedUID = mirrorUID
-			}
-			if m.needsUpdate(types.UID(syncedUID), status) {
-				updatedStatuses = append(updatedStatuses, podStatusSyncRequest{uid, status})
-			} else if m.needsReconcile(uid, status.status) {
-				// Delete the apiStatusVersions here to force an update on the pod status
-				// In most cases the deleted apiStatusVersions here should be filled
-				// soon after the following syncPod() [If the syncPod() sync an update
-				// successfully].
-				delete(m.apiStatusVersions, syncedUID)
-				updatedStatuses = append(updatedStatuses, podStatusSyncRequest{uid, status})
-			}
+			updatedStatuses = append(updatedStatuses, podStatusSyncRequest{podUID: uid, status: status})
 		}
 	}()
 
-	for _, update := range updatedStatuses {
-		klog.V(5).InfoS("Status Manager: syncPod in syncbatch", "podUID", update.podUID)
-		m.syncPod(update.podUID, update.status)
+	// process all pods in priority order
+	var total, update, reconcile int
+	for _, updatedStatus := range updatedStatuses {
+		total++
+		pod, ok := m.podForAPIServer(updatedStatus.podUID)
+		if !ok {
+			continue
+		}
+
+		terminalStatus := m.canBeDeleted(pod, updatedStatus.status.status)
```

Comment:
This value was previously always calculated, but could be invoked twice.  Calculate it only once per pod.